src/pages/studyView/addChartButton/AddChartButton.tsx

```diff
@@ -433,13 +480,27 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         tabs = _.map(
             this.props.store.genericAssayProfileOptionsByType.result,
             (options, type) => {
+                const molecularProfileIdsInType = options.find(
```

Comment:
this code could really use some commenting

Reply:
Sure, I will add some comments here